pkg/storage/cacher.go

```diff
@@ -464,7 +467,15 @@ func (c *cacheWatcher) stop() {
 }
 
 func (c *cacheWatcher) add(event watchCacheEvent) {
-	c.input <- event
+	select {
+	case c.input <- event:
+	case <-time.After(time.Second):
```

Comment:
Yeah, that sounds good. I'll make a PR that includes this commit and does that since you're out for the day.